src/Transport/Topology/MetaModel/EntityInfo.cs

```diff
@@ -22,6 +23,16 @@ public EntityInfo()
 
         protected bool Equals(EntityInfo other)
         {
+            if (Type == EntityType.Subscription)
```

Comment:
I know that @tmasternak @Scooletz and I wrote that code as a quick hack when we debugged the above test. Still on the fence whether we should put that in in this way. Since we have `SubscriptionInfo` should we move and overridden equal into the subscription info?